mscore/resourceManager.cpp

```diff
@@ -29,11 +29,45 @@ ResourceManager::ResourceManager(QWidget *parent) :
       {
       setObjectName("ResourceManager");
       setupUi(this);
+      
+#ifndef NDEBUG
+      QPushButton* check_update = new QPushButton();
+      check_update->setText(tr("Check for Update"));
+      bottom_layout->addWidget(check_update);
+      connect(check_update, SIGNAL(clicked()), this, SLOT(scanPluginUpdate()));
+#endif // !NDEBUG
+
       setWindowFlags(this->windowFlags() & ~Qt::WindowContextHelpButtonHint);
       QDir dir;
       dir.mkpath(dataPath + "/locale");
-      displayExtensions();
-      displayLanguages();
+      workerThreads.setMaxThreadCount(10);
```

Comment:
Just a note here: 
10 is just a number I thought of without much consideration. However, I think a large max thread count is reasonable here, even if it exceeds the CPU thread count. That's because many threads are needed for various jobs, as shown in `PluginWorker`, and a thread doesn't cost much most of the time(it just blocks until the download finishes)

Reply:
probably better to have this as a comment in the code

Reply:
OK, I've mentioned that.